pom.xml

```diff
@@ -108,11 +114,23 @@
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
       <version>${maven.version}</version>
+      <exclusions>
```

Comment:
why the exclusion? I'd expect this change to resolve a conflict, not introduce one that needs to be excluded. Absent a conflict, we should pull in transitive dependencies since we can expect that this dependency depends on that.

Reply:
Because this pr is to "de-plexus". Upcoming change will move these to provided scope, as at runtime these ARE provided.

Reply:
It should only deplexus direct dependencies. Transitive dependencies might be needed and shouldn't be touched.

Reply:
Um, I don't quite understand:
* we are making sure THIS module is plexus-free, to achieve that we are making sure no plexus enters it's classpath (hence the exclusions)
* "transitive dependencies might be needed", due above, this implies that some consumer of this module "might still need" plexus? My stance then is that consumer dependencies are badly set up (using something that is not a dependency), wouldn't in that case https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html trigger an alarm? So when consumer dependencies are changed (version of m-common-a-f upped), at the same time that consumer should be a) de-plexusized as well, or b) have it's deps properly set up.

My 5 cents.

Reply:
We are not trying to make sure there is no plexus in the classpath. We are trying to make sure this plugin does not introduce a dependency on plexus, but this plugin should not attempt to control the dependencies of its own dependencies. Doing so like this risks breaking them in confusing and hard-to-debug ways. If one of this plugin's dependencies introduces plexus, then that needs to be addressed in that dependency, not here.

Exclusions are risky and confusing. They should be reserved only for very well defined and understood conflicts, most often because two different artifacts are otherwise supplying the same classes.

Reply:
ack, then assume exclusions were there to make sure (during dev) that this project does not depend on any Plexus class (directly or indirectly, as in class hierarchy). As this project depends on maven-core and many other maven internals, I added exclusion (as even Maven 4 still brings plexus). So, will remove exclusions as PR builds ok with it in place, hence it proves the point that this project is plexus-free.

Still, my stance is that downstream projects are simply wrong if they **expect** plexus as transitive dependency from this project, which, given this dependency is meant to be consumed as maven plugin or maven extension dependency is already granted even with Maven 4.

IMO, exclusions are "necessary evil", as plexus originates (and will originate) from maven-core, and it will be the LAST project to have plexus removed. But until then, we must make sure all "downstream" projects like this one, are plexus-free, and is prepared to the moment where dependency on maven-core will NOT automatically mean "plexus is here!"